src/CodeReview.Orchestrator/Services/ContainerLogListener.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Buffers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Docker.DotNet;
+using Microsoft.Extensions.Logging;
+
+namespace GodelTech.CodeReview.Orchestrator.Services
+{
+    public sealed class ContainerLogListener : IContainerLogListener
+    {
+        private readonly ILogger<ContainerLogListener> _logger;
+
+        private readonly string _containerId;
+        private readonly MultiplexedStream _stream;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly Task _listenTask;
+
+        public ContainerLogListener(ILogger<ContainerLogListener> logger,
+            long waitTimeoutSeconds,
+            string containerId,
+            MultiplexedStream stream)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _containerId = containerId ?? throw new ArgumentNullException(nameof(containerId));
+            _stream = stream ?? throw new ArgumentNullException(nameof(stream));
+
+            _cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(waitTimeoutSeconds));
+            _listenTask = Task.Factory.StartNew(Listen, _cancellationTokenSource.Token, TaskCreationOptions.LongRunning);
```

Comment:
I would put this line into separate method named  'Start'. To my mind it's better option than start some activity while object is being created.